src/main/java/com/productdock/library/gateway/book/BookService.java

```diff
@@ -13,29 +14,49 @@
 public record BookService(CatalogClient catalogClient, RentalClient rentalClient,
                           InventoryClient inventoryClient, BookDetailsResponseCombiner bookDetailsResponseCombiner) {
 
+    private static final String CLAIM_EMAIL = "email";
+
     @SneakyThrows
     public JsonNode getBookDetailsById(String bookId, String jwtToken) {
+        var jwt = JWTParser.parse(jwtToken);
+        var userId = jwt.getJWTClaimsSet().getClaim(CLAIM_EMAIL).toString();
+
         var bookDtoMono = catalogClient.getBookData(bookId, jwtToken);
         var rentalRecordsDtoMono = rentalClient.getBookRentalRecords(bookId, jwtToken);
         var availableBooksCountMono = inventoryClient.getAvailableBookCopiesCount(bookId, jwtToken);
+        var bookSubscriptionMono = inventoryClient.getBookSubscription(bookId, jwtToken, userId);
 
-        var bookDetailsDtoMono = Mono.zip(bookDtoMono, rentalRecordsDtoMono, availableBooksCountMono).flatMap(tuple -> {
-            var book = bookDetailsResponseCombiner.generateBookDetailsDto(tuple.getT1(), tuple.getT2(), tuple.getT3());
+        var bookDetailsDtoMono = Mono.zip(bookDtoMono, rentalRecordsDtoMono, availableBooksCountMono, bookSubscriptionMono).flatMap(tuple -> {
+            var bookDetailsDto = new BookDetailsDto();
+            bookDetailsDto.setBookDataDto(tuple.getT1());
+            bookDetailsDto.setRentalRecordsDto(tuple.getT2());
+            bookDetailsDto.setAvailableBookCount(tuple.getT3());
+            bookDetailsDto.setBookSubscription(tuple.getT4());
+            var book = bookDetailsResponseCombiner.generateBookDetailsDto(bookDetailsDto);
```

Comment:
This part is repeated as well, `getBookDetailsById` and `getBookDetailsByTitleAndAuthor` can be optimized to avoid code duplication.